src/posix.h

```diff
@@ -0,0 +1,45 @@
+#ifndef BENCHMARK_POSIX_H_
```

Comment:
I'm concerned about naming a header `posix.h` since it's a very generic name. Maybe `posix_support.h` or `benchmark_posix_support.h`